silk-react-components/src/HierarchicalMapping/Components/MappingRule/MappingRule.jsx

```diff
@@ -47,8 +48,12 @@ const MappingRule = React.createClass({
 
     // initilize state
     getInitialState() {
+        const isPasted = sessionStorage.getItem('pastedId') &&
+            (sessionStorage.getItem('pastedId') === this.props.id) &&
+            !sessionStorage.removeItem('pastedId');
```

Comment:
Sure. But I use the same logic to read if it exists, if it's what I want, and if those two are True to remove if from the Storage in less code

Reply:
Yes I understand you but you are passing down to other components 2 type of values

1. `null`
2. `boolean`

It's not good practice that because I wrote that comment, also executing one function multiple time is not a goo practice